src/engine/event/PlayerMove.ts

```diff
@@ -0,0 +1,90 @@
+import EventAbstract from './EventAbstract';
+import { Effect, EventOptions, Validator } from './types';
+import { Direction } from '../../models/Direction';
+import { GameStatus } from '../../models/Game';
+import { Position } from '../../models/Position';
+import { selectGameStatus } from '../../store/reducers/game';
+import { selectMapTiles } from '../../store/reducers/map';
+import { selectPlayerPosition } from '../../store/reducers/player';
+import playerMove from '../../store/thunks/playerMove';
+
+export interface PlayerMoveOptions {
+  direction: Direction;
+}
+type Options = EventOptions & PlayerMoveOptions;
+
+const getPosition = (playerPosition: Position, direction: Direction): Position => {
+  const position: Position = { left: playerPosition.left, top: playerPosition.top };
+  switch (direction) {
+    case 'down':
+      position.top += 1;
+      break;
+    case 'left':
+      position.left -= 1;
+      break;
+    case 'right':
+      position.left += 1;
+      break;
+    case 'up':
+      position.top -= 1;
+      break;
+    default:
+      break;
+  }
+  return position;
+};
+
+const validateGameStatusIdle: Validator<Options> = (options) => {
+  const status = selectGameStatus(options.getState());
+  return status === GameStatus.GAME_STATUS_IDLE;
+};
+
+const validateInMapBounds: Validator<Options> = (options) => {
+  const state = options.getState();
+  const playerPosition = selectPlayerPosition(state);
+  const position = getPosition(playerPosition, options.direction);
+  if (position.left < 0 || position.top < 0) {
+    return false;
+  }
+  const tiles = selectMapTiles(state);
+  if (position.top > tiles.length) {
+    return false;
+  }
+  if (tiles[0] && position.left > tiles[0].length) {
```

Comment:
```suggestion
  if (!tiles[0] || position.left > tiles[0].length) {
```

This will fail faster, and in more scenarios.